### PR TITLE
Book components refactored

### LINKED ISSUE #461: Refactor all Book components
## ✨ Feature request
Use composition api instead

### 📜 Acceptance Criteria
- [x] BookCondition refactored
- [x] BookForm refactored
- [x] BookList refactored

### PR DESCRIPTION
**Pull request by te4-henrik-stahl**
## ⚙️ What issue did you implement or fix?
Fixes #461

## 📜 Summary
- Refactored BookFormComponent
- Refactored BookListComponent
- Refactored BookConditionComponent

## 📝 Type of change

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Te4-Jacob-sandstrom — CHANGES_REQUESTED** [review]
```ERROR in /home/runner/work/Yabs/Yabs/frontend/src/components/__tests__/BookFormComponent.spec.ts(12,28):
12:28 'BookFormComponent' refers to a value, but is being used as a type here.
    10 |   
    11 |   it('calls booksmodule with correct values', () => {
  > 12 |     const wrapper: Wrapper<BookFormComponent> = factory(BookFormComponent);
       |                            ^
    13 | 
    14 |     wrapper.find('[data-jest=\'barcode\']').setValue('1234');
    15 |     wrapper.find('[data-jest=\'condition\']').setValue('ok');```

Ändra <BookFormComponent> till <Vue> på rad 12 i BookFormComponent.spec.ts

**Review by Te4-Jacob-sandstrom — CHANGES_REQUESTED** [review]
Why have you changed things in AdminLoansView and AdminTitlesView? Are they related to this pull request?
Also please fix the merge conflicts

File: frontend/src/components/BookConditionComponent.vue [hunk 1]
```diff
@@ -32,34 +32,52 @@
...
+import { defineComponent } from '@vue/composition-api';
 
-@Component
-export default class BookStatusComponent extends Vue {
-  @Prop({default: 'No Data'}) private title!: string;
-  @Prop({default: 0}) private title_id!: number; //eslint-disable-line camelcase
-  @Prop({default: ''}) private barcode!: string;
-  @Prop({default: 'Unavailable'}) private status!: string;
-  @Prop({default: ''}) private note!: string;
+interface BookConditionProps {
```
Comment (te4-alexander-kjellberg1) [inline frontend/src/components/BookConditionComponent.vue:39]: is this needed? I cannot see what it adds

File: frontend/src/components/BookConditionComponent.vue [hunk 2]
```diff
@@ -32,34 +32,52 @@
...
+    title_id:{type:Number, default:0}, //eslint-disable-line camelcase
+    barcode:{type:String, default:''},
+    status:{type:String, default:'Unavailable'},
+    note:{type:String, default:''}
+  },
+  setup(props:BookConditionProps, ctx:Object){
 
+    const statuses: string [] = ['Damaged', 'OK', 'Other'];
+    
+    console.log(props);
```
Comment (te4-alexander-kjellberg1) [inline frontend/src/components/BookConditionComponent.vue:60]: is this needed?

**Pull request closed without merging**